packages/library-base/src/commonMain/kotlin/io/realm/DynamicMutableRealmObject.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+/**
+ * A **dynamic mutable realm object** gives access and possibility to update the data of the realm
+ * objects through a generic string based API instead of the conventional [Realm] API that only
+ * allows access and updates through the properties of the corresponding schema classes supplied in the configuration.
+ */
+interface DynamicMutableRealmObject : DynamicRealmObject {
```

Comment:
While I understand why it would be nice to have mutability as part of the type system, I have my doubts about it. 

Mostly because this approach breaks down as soon as you access any Object or List, which feels inconsistent.

If we could find a way to have that distinction in the type system for all types it would feel more natural, but that seems like something we should defer to a v2 of this API. 

So for now, I would probably recommend merging this with DynamicRealmObject.

Reply:
It somehow works for objects as you can just get the mutable ones when requesting objects/lists. The nice thing is that the queries on `DynamicRealm`/`DynamicMutableRealm` and `enumerate` yields the correct types. But I guess it ties together with the below comment ... so lets discuss.

Reply:
I added `getObject` variants according to other comments/discussions and as agreed kept the `Mutable` separation.